docs/data/data-grid/prompt/PromptWithDataSource.js

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import {
+  DataGridPremium,
+  Unstable_GridToolbarPromptControl as GridToolbarPromptControl,
+  GridToolbar,
+} from '@mui/x-data-grid-premium';
+import { mockPromptResolver, useMockServer } from '@mui/x-data-grid-generator';
+import Stack from '@mui/material/Stack';
+import Box from '@mui/material/Box';
+
+function ToolbarWithPromptInput() {
+  return (
+    <Stack flex={1} gap={0.5} sx={{ px: 0.5 }}>
+      <GridToolbar />
+      <Box sx={{ px: 0.5 }}>
+        <GridToolbarPromptControl onPrompt={mockPromptResolver} allowDataSampling />
```

Comment:
Looks like we need a special prompt resolver mock for this demo

Reply:
I have updated mock server to return the same data set as in the other examples
But there is another issue that we have to resolve

once the prompt response is there it triggers couple of updates at once, which triggers a couple of events
data source reacts to all of them separately, so it makes a few requests at once and the last response may not be the last update

Reply:
Isn't this the same issue as https://github.com/mui/mui-x/pull/13878#discussion_r1826240944 ?

Reply:
yes, but apart from addressing the race condition I have added a point to support handling all of the prompt response changes with one additional request